src/sentry/integrations/bitbucket/integration.py

```diff
@@ -96,11 +98,26 @@ def get_repositories(self, query=None):
                 for repo in resp.get("values", [])
             ]
 
-        full_query = (u'name~"%s"' % (query)).encode("utf-8")
-        resp = self.get_client().search_repositories(self.username, full_query)
-        return [
-            {"identifier": i["full_name"], "name": i["full_name"]} for i in resp.get("values", [])
-        ]
+        exact_query = (u'name="%s"' % (query)).encode("utf-8")
+        fuzzy_query = (u'name~"%s"' % (query)).encode("utf-8")
+
+        exact_search_resp = self.get_client().search_repositories(self.username, exact_query)
+        fuzzy_search_resp = self.get_client().search_repositories(self.username, fuzzy_query)
+
+        if len(exact_search_resp.get("values", [])) == 0:
+            return [
+                {"identifier": i["full_name"], "name": i["full_name"]}
+                for i in fuzzy_search_resp.get("values", [])
+            ]
```

Comment:
This feels like repeated logic. How about something like this:
1. Get all the results from the fuzzy search and put it in the ordered set
2. Get all the results from the exact search and put it in the ordered set
3. Convert the ordered set to the return type we need